
package MsgsReplayToClient;

import "Common.proto";
import "SimToClient.proto";

message MsgControlReplayInformation {
   required int32                       current_tick      = 1;
   required Common.MsgDateTime          initial_date_time = 2;
   required Common.MsgDateTime          date_time         = 3;
   required int32                       tick_duration     = 4;
   required int32                       time_factor       = 5;
   required Common.EnumSimulationState  status            = 6;
   required int32                       tick_count        = 7;
}

message MsgControlSkipToTickAck {
   required int32                                tick       = 1;
   required MsgsSimToClient.ControlAck.ErrorCode error_code = 2;
}

message MsgReplayToClient {
    message Content {
        optional MsgControlReplayInformation                   control_replay_information      = 1;
        optional MsgControlSkipToTickAck                       control_skip_to_tick_ack        = 2;
        optional MsgsSimToClient.MsgControlStopAck             control_stop_ack                = 3;
        optional MsgsSimToClient.MsgControlPauseAck            control_pause_ack               = 4;
        optional MsgsSimToClient.MsgControlResumeAck           control_resume_ack              = 5;
        optional MsgsSimToClient.MsgControlChangeTimeFactorAck control_change_time_factor_ack  = 6;
    }
    optional int32   context = 1 [default = 0];
    required Content message = 2;
}
